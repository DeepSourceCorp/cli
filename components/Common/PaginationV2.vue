<template>
  <div class="flex items-center justify-center gap-x-2">
    <p class="mr-2 text-sm text-vanilla-400">
      {{ start }} - {{ end }} <span class="text-slate-200">of</span> {{ totalCount }}
    </p>

    <z-divider direction="vertical" margin="m-0" />

    <button
      class="h-6 w-6 rounded-sm text-slate-200 hover:bg-ink-200"
      @click="backActionClickHandler"
    >
      <z-icon :color="backActionIconColor" icon="chevron-left" class="mx-auto" />
    </button>

    <button
      class="h-6 w-6 rounded-sm text-slate-200 hover:bg-ink-200"
      @click="forwardActionClickHandler"
    >
      <z-icon :color="forwardActionIconColor" icon="chevron-right" class="mx-auto" />
    </button>
  </div>
</template>

<script lang="ts">
import { ZButton, ZDivider, ZIcon } from '@deepsource/zeal'
import { Component, ModelSync, Prop, Vue } from 'nuxt-property-decorator'

@Component({
  name: 'PaginationV2',
  components: {
    ZButton,
    ZDivider,
    ZIcon
  }
})
export default class PaginationV2 extends Vue {
  @ModelSync('pageNumber', 'change', { type: Number })
  readonly modelValue: number

  @Prop({ required: true, type: Number })
  perPageCount: number

  @Prop({ required: true, type: Number })
  totalCount: number

  get backActionIconColor() {
    return this.disableBackAction ? 'slate-200' : 'vanilla-400'
  }

  get forwardActionIconColor() {
    return this.disableForwardAction ? 'slate-200' : 'vanilla-400'
  }

  get disableBackAction() {
    return this.modelValue <= 1
  }

  get disableForwardAction() {
    return this.modelValue >= Math.ceil(this.totalCount / this.perPageCount)
  }

  get end() {
    const count = this.perPageCount * this.modelValue
    return count > this.totalCount ? this.totalCount : count
  }

  get start() {
    return this.perPageCount * (this.modelValue - 1) + 1
  }

  backActionClickHandler() {
    return this.disableBackAction ? false : this.$emit('change', this.modelValue - 1)
  }

  forwardActionClickHandler() {
    return this.disableForwardAction ? false : this.$emit('change', this.modelValue + 1)
  }
}
</script>
