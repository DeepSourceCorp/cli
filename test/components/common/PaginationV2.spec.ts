import { render } from '@testing-library/vue'

import PaginationV2 from '~/components/Common/PaginationV2.vue'

describe('[[ PaginationV2 ]]', () => {
  const propsData = {
    pageNumber: 1,
    perPageCount: 30,
    totalCount: 250
  }

  const stubs = {
    ZButton: true,
    ZDivider: true,
    ZIcon: true
  }

  it('renders the new pagination pattern', () => {
    const { html } = render(PaginationV2, { propsData, stubs })

    expect(html()).toMatchSnapshot()
  })
})
