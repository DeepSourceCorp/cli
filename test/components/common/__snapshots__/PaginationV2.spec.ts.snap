// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ PaginationV2 ]] renders the new pagination pattern 1`] = `
"<div class=\\"flex items-center justify-center gap-x-2\\">
  <p class=\\"mr-2 text-sm text-vanilla-400\\">
    1 - 30 <span class=\\"text-slate-200\\">of</span> 250
  </p>
  <zdivider-stub direction=\\"vertical\\" margin=\\"m-0\\" color=\\"ink-200\\"></zdivider-stub> <button class=\\"h-6 w-6 rounded-sm cursor-not-allowed text-slate-200\\">
    <zicon-stub icon=\\"chevron-left\\" size=\\"small\\" position=\\"\\" color=\\"current\\" class=\\"mx-auto\\"></zicon-stub>
  </button> <button class=\\"h-6 w-6 rounded-sm text-vanilla-400 hover:bg-ink-200\\">
    <zicon-stub icon=\\"chevron-right\\" size=\\"small\\" position=\\"\\" color=\\"current\\" class=\\"mx-auto\\"></zicon-stub>
  </button>
</div>"
`;
